Add support for the note field on friend requests (#579)
discord-userdoccers/discord-userdoccers: pages/resources/relationships.mdx

| origin_application_id? | ?snowflake                                  | The ID of the application that created the relationship                                                                      |
| since?                 | ISO8601 timestamp                           | When the user requested a relationship                                                                                       |
| has_played_game? ^1^   | boolean                                     | Whether the target user has authorized the same application the current user's session is associated with                    |

> | note                   | ?string                                     | The personalised note the user wrote when sending the friend request                                                         |

Reviewer: ```suggestion
| note?                  | ?string                                     | The personalized note the user wrote when sending the friend request                                                         |
```
Is it actually null?

###### JSON Params

| Field             | Type    | Description                                                                            |
| ----------------- | ------- | -------------------------------------------------------------------------------------- |
| username          | string  | The username of the user to send a friend request to                                   |
| discriminator ^1^ | ?string | The discriminator of the user to send a friend request to                              |

> | note              | ?string | A personalised note to send with the friend request. Will be sent to the user as a DM. |

Reviewer 2: Won't be sent to the DM channel until they accept the friend request, maybe reword this to say "Will be shown to the recipient"?

> | note              | ?string | A personalised note to send with the friend request. Will be sent to the user as a DM. |

Reviewer: ```suggestion
| note?             | ?string | A personalized note to send with the friend request |
```
No period in the end (also `personalised` (british) -> `personalized` (american))

| ----------------- | ------- | --------------------------------------------------------- |
| username          | string  | The username of the user to send a friend request to      |
| discriminator ^1^ | ?string | The discriminator of the user to send a friend request to |

> | note?             | ?string | A personalized note to send with the friend request       |

Reviewer 3: ```suggestion
| note?             | ?string | A personalized note to send with the friend request (max 120 characters)      |
```

| type?                     | integer | The [relationship type](#relationship-type) to create (defaults to -1, which accepts an existing or creates a new friend request) |
| from_friend_suggestion?   | boolean | Whether the relationship was created from a friend suggestion (default false)                                                     |
| confirm_stranger_request? | boolean | Whether the user consents to accepting a stranger's friend request (default false)                                                |

> | note                      | ?string | A personalised note to send with the friend request. Will be sent to the user as a DM.                                            |

Reviewer: ```suggestion
| note?                     | ?string | A personalised note to send with the friend request                                            |
```

| type?                     | integer | The [relationship type](#relationship-type) to create (defaults to -1, which accepts an existing or creates a new friend request) |
| from_friend_suggestion?   | boolean | Whether the relationship was created from a friend suggestion (default false)                                                     |
| confirm_stranger_request? | boolean | Whether the user consents to accepting a stranger's friend request (default false)                                                |

> | note?                     | ?string | A personalized note to send with the friend request                                                                               |

Reviewer 3: ```suggestion
| note?                     | ?string | A personalized note to send with the friend request (max 120 characters)                                                                               |
```
